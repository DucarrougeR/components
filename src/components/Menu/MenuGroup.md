`MenuGroup` is a way to group together similar collections of `MenuItem`s.

It supports an optional `label` prop to place a heading above the `MenuItem`s. When placed within a scrolling container the label will stick to the top edge while the MenuGroup is visible. Look at the "Menu + Drawer" example above and try scrolling.

`labelProps` and `labelStyles` can be used to apply styling to the label.

```js
import { Menu } from './Menu'
import { MenuGroup } from './MenuGroup'
import { MenuItem } from './MenuItem'
;<Menu groupDividers={false}>
  <MenuGroup>
    <MenuItem icon="LogoRings">Looker</MenuItem>
    <MenuItem icon="Validate">Validate</MenuItem>
    <MenuItem icon="ChartPie">Pizza!</MenuItem>
  </MenuGroup>
  <MenuGroup label="Cheeses">
    <MenuItem icon="FavoriteOutline">Cheddar</MenuItem>
    <MenuItem icon="FavoriteOutline">Cheddar</MenuItem>
    <MenuItem icon="FavoriteOutline">Swiss</MenuItem>
  </MenuGroup>
  <MenuGroup label="Cheeses">
    <MenuItem icon="FavoriteOutline">Cheddar</MenuItem>
    <MenuItem icon="FavoriteOutline">Cheddar</MenuItem>
    <MenuItem icon="FavoriteOutline">Swiss</MenuItem>
  </MenuGroup>
</Menu>
```

You can disable the shadows for `MenuGroup` `label`'s by setting the  `groupLabelShadow` prop to false either at the `Menu` level or individually on each `MenuGroup`

```js
import { Menu } from './Menu'
import { MenuGroup } from './MenuGroup'
import { MenuItem } from './MenuItem'
;<Menu groupLabelShadow={false}>
  <MenuGroup>
    <MenuItem icon="LogoRings">Looker</MenuItem>
    <MenuItem icon="Validate">Validate</MenuItem>
    <MenuItem icon="ChartPie">Pizza!</MenuItem>
  </MenuGroup>
  <MenuGroup label="Cheeses">
    <MenuItem icon="FavoriteOutline">Cheddar</MenuItem>
    <MenuItem icon="FavoriteOutline">Cheddar</MenuItem>
    <MenuItem icon="FavoriteOutline">Swiss</MenuItem>
  </MenuGroup>
  <MenuGroup label="Cheeses">
    <MenuItem icon="FavoriteOutline">Cheddar</MenuItem>
    <MenuItem icon="FavoriteOutline">Cheddar</MenuItem>
    <MenuItem icon="FavoriteOutline">Swiss</MenuItem>
  </MenuGroup>
</Menu>
