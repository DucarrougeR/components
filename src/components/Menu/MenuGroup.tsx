import * as React from 'react'
import { palette, styled } from '../../style'
import { Box, BoxPropsWithout } from '../Box'
import { Heading, HeadingProps } from '../Heading'
import { MenuContext, MenuContextProps } from './MenuContext'
import { MenuItemCustomizationProps } from './MenuItem'

export interface MenuGroupProps
  extends BoxPropsWithout<HTMLDivElement, 'label'>,
    MenuContextProps {
  label?: React.ReactNode
  labelProps?: HeadingProps
  labelStyles?: React.CSSProperties
  canActivate?: boolean
  customizationProps?: MenuItemCustomizationProps
}

const Internal: React.SFC<MenuGroupProps> = ({
  children,
  label,
  labelProps,
  labelStyles,
  ...props
}) => {
  const groupCanActivate = props.canActivate
  const suppliedCustomizations = props.customizationProps
  delete props.canActivate // Prevent canActivate from being applied to Heading component
  delete props.customizationProps // Prevent customizationProps from being applied to Heading component

  const labelComponent = label && (
    <Heading
      bg={palette.white}
      fontSize="xsmall"
      is="h2"
      px="medium"
      py="small"
      transform="upper"
      position="sticky"
      top="0"
      fontWeight="semiBold"
      boxShadow={`0 4px 8px -2px ${palette.charcoal200}`}
      {...labelProps}
      style={labelStyles}
      zIndex={1}
    >
      {label}
    </Heading>
  )

  return (
    <MenuContext.Consumer>
      {({ canActivate, customizationProps }) => (
        <MenuContext.Provider
          value={{
            canActivate:
              groupCanActivate !== undefined ? groupCanActivate : canActivate,
            customizationProps:
              suppliedCustomizations !== undefined
                ? suppliedCustomizations
                : customizationProps,
          }}
        >
          <Box {...props}>
            {labelComponent}
            {children}
          </Box>
        </MenuContext.Provider>
      )}
    </MenuContext.Consumer>
  )
}

export const MenuGroup = styled(Internal)``
