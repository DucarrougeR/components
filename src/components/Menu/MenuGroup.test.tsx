import 'jest-styled-components'
import * as React from 'react'
import { assertSnapshot } from '../../../test/utils/snapshot'

import { palette } from '../../style'
import { Box } from '../Box'
import { MenuGroup } from './MenuGroup'
import { MenuItem } from './MenuItem'

test('MenuGroup', () => {
  assertSnapshot(
    <MenuGroup>
      <MenuItem>who!</MenuItem>
    </MenuGroup>
  )
})

test('MenuGroup - canActivate', () => {
  assertSnapshot(
    <MenuGroup canActivate>
      <MenuItem>what?</MenuItem>
      <MenuItem>who?</MenuItem>
      <MenuItem>where?</MenuItem>
    </MenuGroup>
  )
})

test('MenuGroup - label', () => {
  assertSnapshot(
    <MenuGroup label="Questions">
      <MenuItem>what?</MenuItem>
      <MenuItem>who?</MenuItem>
      <MenuItem>where?</MenuItem>
    </MenuGroup>
  )
})

test('MenuGroup - JSX label', () => {
  assertSnapshot(
    <MenuGroup label={<Box>Questions</Box>}>
      <MenuItem>what?</MenuItem>
      <MenuItem>who?</MenuItem>
      <MenuItem>where?</MenuItem>
    </MenuGroup>
  )
})

test('MenuGroup - labelProps & labelStyles', () => {
  assertSnapshot(
    <MenuGroup
      label="Questions"
      labelStyles={{ color: 'rebeccapurple' }}
      labelProps={{ fontSize: 'xxlarge' }}
    >
      <MenuItem>what?</MenuItem>
      <MenuItem>who?</MenuItem>
      <MenuItem>where?</MenuItem>
    </MenuGroup>
  )
})

test('MenuGrouo - menu customization', () => {
  const menuCustomizations =
    // tslint:disable:object-literal-sort-keys
    {
      bg: palette.purple500,
      color: palette.purple200,
      icon: {
        color: palette.purple300,
        size: 20,
      },
      marker: {
        color: palette.purple900,
        size: 10,
      },
      hover: {
        bg: palette.purple700,
        color: palette.white,
        icon: {
          color: palette.purple100,
        },
      },
      current: {
        bg: palette.purple200,
        color: palette.purple900,
        icon: {
          color: palette.purple500,
        },
      },
      activated: {
        color: palette.blue500,
      },
      // tslint:disable:enable-literal-sort-keys
    }

  const menu = assertSnapshot(
    <MenuGroup customizationProps={menuCustomizations}>
      <MenuItem icon="LogoRings">Looker</MenuItem>
      <MenuItem icon="Validate">Validate</MenuItem>
      <MenuItem icon="ChartPie">Pizza!</MenuItem>
    </MenuGroup>
  )

  expect(menu).toMatchSnapshot()
})
