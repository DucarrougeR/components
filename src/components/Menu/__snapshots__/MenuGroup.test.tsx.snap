// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuGrouo - menu customization 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin: 0rem;
  padding: 0rem;
}

.c6 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: none;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #BFB2FF;
  background-color: #4F2ABA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 0.875rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

.c4:active {
  color: #0087e1;
}

.c8 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
  margin-right: 0.5rem;
  width: 20px;
}

.c1 {
  list-style-type: none;
}

.c9:hover .c7 {
  color: #262D33;
}

.c9 .c7 {
  -webkit-transition: color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c9 .c7 {
  color: #C1C6CC;
}

.c3 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c3:hover {
  background: #341C7A;
  color: #FFFFFF;
}

.c3:hover .c7 {
  color: #E8E5FF;
}

.c3 .c7 {
  -webkit-transition: color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c3 .c7 {
  color: #9785F2;
}

.c3:focus-within {
  box-shadow: 0 0 0.25rem 0.125rem #49a9f2;
  outline: 'none';
  z-index: 1;
}

.c5 {
  outline: none;
  background: transparent;
  color: inherit;
  text-align: left;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<li
  className="c0"
>
  <ul
    className="c1 c2"
  >
    <li
      className="c3 c4"
    >
      <button
        className="c5 c6"
        role="menuitem"
      >
        <div
          className="c7 c8"
        >
          <svg
            fill="currentColor"
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
          >
            <g
              clipPath="url(#Logo_Rings_svg__clip0)"
              fill="currentColor"
            >
              <path
                d="M13.913 11.36a6.339 6.339 0 0 0-3.199.864l.914 1.207a4.841 4.841 0 0 1 2.286-.572 4.88 4.88 0 0 1 4.875 4.875 4.88 4.88 0 0 1-4.875 4.875 4.88 4.88 0 0 1-4.875-4.875c0-1.323.531-2.523 1.39-3.402l-.913-1.206a6.354 6.354 0 0 0-1.977 4.608 6.382 6.382 0 0 0 6.375 6.375 6.382 6.382 0 0 0 6.375-6.375 6.382 6.382 0 0 0-6.375-6.375v.001zM9.785 8.735c0 .59-.198 1.132-.527 1.57l.927 1.224a4.1 4.1 0 0 0 .319-5.201l-1.1 1.052c.24.396.38.858.38 1.354v.001zM7.16 11.361a2.629 2.629 0 0 1-2.625-2.625A2.628 2.628 0 0 1 7.16 6.111c.405 0 .79.095 1.132.26l1.13-1.08a4.1 4.1 0 0 0-2.262-.68 4.13 4.13 0 0 0-4.125 4.125 4.13 4.13 0 0 0 4.125 4.125c.655 0 1.273-.158 1.824-.43l-.93-1.23c-.28.102-.58.16-.894.16zM11.285 3.111c0-.827.673-1.5 1.5-1.5s1.5.673 1.5 1.5a1.502 1.502 0 0 1-1.92 1.44l-1.158 1.107c.46.285 1 .453 1.578.453 1.654 0 3-1.346 3-3s-1.346-3-3-3-3 1.346-3 3c0 .532.141 1.031.384 1.464l1.16-1.11a1.481 1.481 0 0 1-.044-.355v.001z"
              />
              <path
                d="M10.184 11.53a4.143 4.143 0 0 1-1.2.9l.528.696a6.457 6.457 0 0 1 1.2-.902l-.527-.695v.001zM10.504 6.33l.703-.673a3.018 3.018 0 0 1-1.038-1.083l-.747.715c.422.278.788.631 1.083 1.041h-.001z"
              />
              <path
                d="M11.625 13.434l-.914-1.207a6.37 6.37 0 0 0-1.2.902l.914 1.206a4.93 4.93 0 0 1 1.199-.901zM9.407 7.381l1.099-1.052a4.153 4.153 0 0 0-1.083-1.04L8.293 6.37a2.64 2.64 0 0 1 1.113 1.011h.001zM10.186 11.53l-.927-1.223a2.63 2.63 0 0 1-1.204.894l.93 1.229c.454-.225.86-.531 1.2-.9zM11.21 5.659l1.158-1.108a1.502 1.502 0 0 1-1.036-1.084l-1.16 1.11c.249.443.608.815 1.038 1.083v-.001z"
              />
            </g>
            <defs>
              <clipPath
                id="Logo_Rings_svg__clip0"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="#fff"
                />
              </clipPath>
            </defs>
          </svg>
        </div>
        Looker
      </button>
    </li>
    <li
      className="c3 c4"
    >
      <button
        className="c5 c6"
        role="menuitem"
      >
        <div
          className="c7 c8"
        >
          <svg
            fill="currentColor"
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
          >
            <path
              d="M11.749.09L2 4.423v6.499c0 6.01 4.16 11.633 9.749 12.998 5.589-1.365 9.749-6.986 9.749-12.998v-6.5L11.749.09zM9.582 17.42l-4.333-4.333 1.527-1.527 2.805 2.794 7.138-7.138 1.527 1.538-8.666 8.665h.002z"
              fill="currentColor"
            />
          </svg>
        </div>
        Validate
      </button>
    </li>
    <li
      className="c3 c4"
    >
      <button
        className="c5 c6"
        role="menuitem"
      >
        <div
          className="c7 c8"
        >
          <svg
            fill="currentColor"
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
          >
            <path
              d="M11 2v20c-5.07-.5-9-4.79-9-10s3.93-9.5 9-10zm2.03 0v8.99H22c-.47-4.74-4.24-8.52-8.97-8.99zm0 11.01V22c4.74-.47 8.5-4.25 8.97-8.99h-8.97z"
              fill="currentColor"
            />
          </svg>
        </div>
        Pizza!
      </button>
    </li>
  </ul>
</li>
`;

exports[`MenuGrouo - menu customization 2`] = `undefined`;

exports[`MenuGroup - JSX label 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin: 0rem;
  padding: 0rem;
}

.c6 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #4C535B;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 0.875rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

.c6:active {
  color: #0087e1;
}

.c8 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: none;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  box-shadow: 0 4px 8px -2px #DEE1E5;
  background-color: #FFFFFF;
  font-size: 0.75rem;
  font-weight: 600;
  line-height: 1rem;
  position: -webkit-sticky;
  position: sticky;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  top: 0;
  z-index: 2;
}

.c1 {
  text-transform: uppercase;
}

.c3 {
  list-style-type: none;
}

.c5 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c5:hover {
  background: #F5F6F7;
  color: #262D33;
}

.c5:hover .c9 {
  color: #262D33;
}

.c5 .c9 {
  -webkit-transition: color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c5 .c9 {
  color: #C1C6CC;
}

.c5:focus-within {
  box-shadow: 0 0 0.25rem 0.125rem #49a9f2;
  outline: 'none';
  z-index: 1;
}

.c7 {
  outline: none;
  background: transparent;
  color: inherit;
  text-align: left;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<li
  className="c0"
>
  <h2
    className="c1 c2"
  >
    <div
      className="c0"
    >
      Questions
    </div>
  </h2>
  <ul
    className="c3 c4"
  >
    <li
      className="c5 c6"
    >
      <button
        className="c7 c8"
        role="menuitem"
      >
        what?
      </button>
    </li>
    <li
      className="c5 c6"
    >
      <button
        className="c7 c8"
        role="menuitem"
      >
        who?
      </button>
    </li>
    <li
      className="c5 c6"
    >
      <button
        className="c7 c8"
        role="menuitem"
      >
        where?
      </button>
    </li>
  </ul>
</li>
`;

exports[`MenuGroup - canActivate 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin: 0rem;
  padding: 0rem;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #4C535B;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 0.875rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

.c4:active {
  color: #0087e1;
}

.c6 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: none;
}

.c7 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  width: 1.5rem;
}

.c1 {
  list-style-type: none;
}

.c3 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c3:hover {
  background: #F5F6F7;
  color: #262D33;
}

.c3:hover .c8 {
  color: #262D33;
}

.c3 .c8 {
  -webkit-transition: color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c3 .c8 {
  color: #C1C6CC;
}

.c3:focus-within {
  box-shadow: 0 0 0.25rem 0.125rem #49a9f2;
  outline: 'none';
  z-index: 1;
}

.c5 {
  outline: none;
  background: transparent;
  color: inherit;
  text-align: left;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<li
  className="c0"
>
  <ul
    className="c1 c2"
  >
    <li
      className="c3 c4"
    >
      <button
        className="c5 c6"
        role="menuitemradio"
      >
        <div
          className="c7"
        />
        what?
      </button>
    </li>
    <li
      className="c3 c4"
    >
      <button
        className="c5 c6"
        role="menuitemradio"
      >
        <div
          className="c7"
        />
        who?
      </button>
    </li>
    <li
      className="c3 c4"
    >
      <button
        className="c5 c6"
        role="menuitemradio"
      >
        <div
          className="c7"
        />
        where?
      </button>
    </li>
  </ul>
</li>
`;

exports[`MenuGroup - label 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin: 0rem;
  padding: 0rem;
}

.c6 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #4C535B;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 0.875rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

.c6:active {
  color: #0087e1;
}

.c8 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: none;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  box-shadow: 0 4px 8px -2px #DEE1E5;
  background-color: #FFFFFF;
  font-size: 0.75rem;
  font-weight: 600;
  line-height: 1rem;
  position: -webkit-sticky;
  position: sticky;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  top: 0;
  z-index: 2;
}

.c1 {
  text-transform: uppercase;
}

.c3 {
  list-style-type: none;
}

.c5 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c5:hover {
  background: #F5F6F7;
  color: #262D33;
}

.c5:hover .c9 {
  color: #262D33;
}

.c5 .c9 {
  -webkit-transition: color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c5 .c9 {
  color: #C1C6CC;
}

.c5:focus-within {
  box-shadow: 0 0 0.25rem 0.125rem #49a9f2;
  outline: 'none';
  z-index: 1;
}

.c7 {
  outline: none;
  background: transparent;
  color: inherit;
  text-align: left;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<li
  className="c0"
>
  <h2
    className="c1 c2"
  >
    Questions
  </h2>
  <ul
    className="c3 c4"
  >
    <li
      className="c5 c6"
    >
      <button
        className="c7 c8"
        role="menuitem"
      >
        what?
      </button>
    </li>
    <li
      className="c5 c6"
    >
      <button
        className="c7 c8"
        role="menuitem"
      >
        who?
      </button>
    </li>
    <li
      className="c5 c6"
    >
      <button
        className="c7 c8"
        role="menuitem"
      >
        where?
      </button>
    </li>
  </ul>
</li>
`;

exports[`MenuGroup - labelProps & labelStyles 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin: 0rem;
  padding: 0rem;
}

.c6 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #4C535B;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 0.875rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

.c6:active {
  color: #0087e1;
}

.c8 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: none;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  box-shadow: 0 4px 8px -2px #DEE1E5;
  background-color: #FFFFFF;
  font-size: 1.5625rem;
  font-weight: 600;
  line-height: 2.5rem;
  position: -webkit-sticky;
  position: sticky;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  top: 0;
  z-index: 2;
}

.c1 {
  text-transform: uppercase;
}

.c3 {
  list-style-type: none;
}

.c5 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c5:hover {
  background: #F5F6F7;
  color: #262D33;
}

.c5:hover .c9 {
  color: #262D33;
}

.c5 .c9 {
  -webkit-transition: color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c5 .c9 {
  color: #C1C6CC;
}

.c5:focus-within {
  box-shadow: 0 0 0.25rem 0.125rem #49a9f2;
  outline: 'none';
  z-index: 1;
}

.c7 {
  outline: none;
  background: transparent;
  color: inherit;
  text-align: left;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<li
  className="c0"
>
  <h2
    className="c1 c2"
    style={
      Object {
        "color": "rebeccapurple",
      }
    }
  >
    Questions
  </h2>
  <ul
    className="c3 c4"
  >
    <li
      className="c5 c6"
    >
      <button
        className="c7 c8"
        role="menuitem"
      >
        what?
      </button>
    </li>
    <li
      className="c5 c6"
    >
      <button
        className="c7 c8"
        role="menuitem"
      >
        who?
      </button>
    </li>
    <li
      className="c5 c6"
    >
      <button
        className="c7 c8"
        role="menuitem"
      >
        where?
      </button>
    </li>
  </ul>
</li>
`;

exports[`MenuGroup 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin: 0rem;
  padding: 0rem;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #4C535B;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 0.875rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

.c4:active {
  color: #0087e1;
}

.c6 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: none;
}

.c1 {
  list-style-type: none;
}

.c3 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: background 150ms cubic-bezier(0.86,0,0.07,1), color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c3:hover {
  background: #F5F6F7;
  color: #262D33;
}

.c3:hover .c7 {
  color: #262D33;
}

.c3 .c7 {
  -webkit-transition: color 150ms cubic-bezier(0.86,0,0.07,1);
  transition: color 150ms cubic-bezier(0.86,0,0.07,1);
}

.c3 .c7 {
  color: #C1C6CC;
}

.c3:focus-within {
  box-shadow: 0 0 0.25rem 0.125rem #49a9f2;
  outline: 'none';
  z-index: 1;
}

.c5 {
  outline: none;
  background: transparent;
  color: inherit;
  text-align: left;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<li
  className="c0"
>
  <ul
    className="c1 c2"
  >
    <li
      className="c3 c4"
    >
      <button
        className="c5 c6"
        role="menuitem"
      >
        who!
      </button>
    </li>
  </ul>
</li>
`;
